observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -0,0 +1,213 @@
+            MICROMETER_API_MODULE,
+            "io.opentelemetry.otlp",
+            "io.micrometer"
+    };
+
+    public static final SimpleAttributeDefinition ENDPOINT = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.ENDPOINT, ModelType.STRING)
+            .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    public static final SimpleAttributeDefinition STEP = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.STEP, ModelType.LONG, true)
+            .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
+            .setDefaultValue(new ModelNode(TimeUnit.MINUTES.toSeconds(1)))
+            .setMeasurementUnit(MeasurementUnit.SECONDS)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final StringListAttributeDefinition EXPOSED_SUBSYSTEMS =
+            new StringListAttributeDefinition.Builder("exposed-subsystems")
+                    .setDefaultValue(ModelNode.fromJSONString("[\"*\"]"))
+                    .setRequired(false)
+                    .setRestartAllServices()
+                    .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = {
+            EXPOSED_SUBSYSTEMS,
+            ENDPOINT,
+            STEP
+    };
+
+    private final AtomicReference<MicrometerDeploymentConfiguration> deploymentConfig = new AtomicReference<>();
+
+    @Override
+    public ManagementResourceRegistration register(SubsystemRegistration parent,
+                                                   ManagementResourceRegistrationContext context) {
+        ManagementResourceRegistration registration =
+                parent.registerSubsystemModel(ResourceDefinition.builder(ResourceRegistration.of(PATH), RESOLVER).build());
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(RESOLVER)
+                .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+                .addCapability(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY)
+                .addAttributes(List.of(ENDPOINT, STEP, EXPOSED_SUBSYSTEMS))
+                .withDeploymentChainContributor(target -> {
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, DEPENDENCIES, DEPENDENCIES_MICROMETER,
+                            new MicrometerDependencyProcessor());
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, POST_MODULE, POST_MODULE_MICROMETER,
+                            new MicrometerDeploymentProcessor(deploymentConfig.get()));
+                })
+                .withAddOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
+                .withRemoveOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
```

Comment:
N.B. This will become redundant following https://github.com/wildfly/wildfly-core/pull/5989